## Задача 1
Даны значения зарплат из выборки выпускников:
100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150



In [52]:
a = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150]
b = len(a)
b

21

In [53]:
c = sum(a)
c

1536

In [54]:
d = (c/b)
d

73.14285714285714

In [55]:
new_list = [((i - d)**2)/b for i in a]
z = sum(new_list)**0.5
z = float('{:.8f}'.format(z))
z

46.20686994

## Задача 2
Найдите число выбросов в выборке из задачи 1. Для определения выбросов используйте методику как при построении “усов” в boxplot, однако, как и в задаче 1, пользоваться можно только встроенными функциями и структурами данных.


In [ ]:
Сортируем список

In [60]:
a.sort()
a

[17,
 24,
 25,
 30,
 33,
 45,
 55,
 57,
 65,
 65,
 70,
 75,
 75,
 77,
 80,
 84,
 89,
 90,
 100,
 150,
 230]

Находим медиану

In [68]:
def median(a):
    n= len(a)
    mid = n // 2
    if n % 2 == 1:
        return sorted(a)[mid]
    else:
        return mean(sorted(a)[mid-1:][:2])
mid

10

In [ ]:
Находим квантиль

In [73]:
Q1 = n * 0.25
Q1

5.25

In [75]:
Q2 = n * 0.5
Q2

10.5

In [76]:
Q3 = n * 0.75
Q3

15.75

In [87]:
IQR = Q3 - Q1
IQR

10.5

Находим границы выбросов

In [92]:
qun_1 = Q1 -1.5 * IQR
qun_1

-10.5

In [93]:
qun_2 = Q3 + 1.5 * IQR
qun_2

31.5

Всё что находится за границами < -10.5 и >31.5 считается выбросами

Дальше не понял как делать без библиотек( буду рад если разъясните мне что нужно было сделать)


## Задача 3
В университет на факультет A поступило 100 человек, на факультет B - 90 человек и на факультет C - 45 человек. Вероятность того, что студент с факультета A сдаст первую сессию, равна 0.6. Для студента с факультета B эта вероятность равна 0.7, а для студента с факультета C - 0.8. Случайным образом выбирается студент, сдавший первую сессию. Какое событие наиболее вероятно:

студент учится на факультете A,

студент учится на факультете B,

студент учится на факультете C?


In [102]:
v1 = 100 * 0.6
v1

60.0

In [103]:
v2 = 90 * 0.7
v2

62.99999999999999

In [104]:
v3 = 45 * 0.8
v3

36.0

Ответ: студент с факультета В